### PR TITLE
[wip] coqobjinfo

### PR DESCRIPTION
**Pull request by gares**
This PR is to provide a tool to inspect vo files with an output that is human and machine readable

### CODE REVIEW

File: lib/stateid.ml [hunk 1]
```diff
@@ -40,8 +40,10 @@ type ('a,'b) request = {
   exn_info : t * t;
   stop : t;
   document : 'b;
-  loc : Loc.t option;
+  loc_begin : Loc.t option;
+  loc_end : Loc.t option;
   uuid     : 'a;
-  name     : string
+  name     : string;
+  kernel_name : Names.Constant.t;
```
Comment (ejgallego) [inline lib/stateid.ml]: This may be a good opportunity to move `request` to `Library`?

**Comment by ejgallego** [comment]
> TODO:

A test would be also nice to have.

**Comment by gares** [comment]
@palmskog did you have time to try this one?

**Comment by gares** [comment]
@palmskog ping

**Comment by palmskog** [comment]
@gares we (I) are still interested in finishing this extension - 8.11 is likely a good target - but other commitments are in the way for the next couple of weeks. I explained a bit more in the Gitter chat.

**Comment by palmskog** [comment]
Plan for this PR:
1. rebase + store vio metadata in separate segment @gares
2. add coqobjinfo to PR @ppedrot 
3. only build coqobjinfo with Dune (topbin/dune) @ppedrot 
4. write tests for coqobjinfo, write tests, coq/test-suite/misc - put subdir with shell script and diff (example redirect_printing.sh) @palmskog

**Comment by gares** [comment]
Rebased and cleaned up.
There is still one segment with everything, but I run out of time.
@ppedrot the ball is yours.

**Comment by ppedrot** [comment]
Here is a POC. For now it's just printing the name of the proof, but it can be easily extended from here.

**Comment by ppedrot** [comment]
I went for trivial sexps, but if you want JSON it's enough to add a standalone library somewhere in the Coq repo.

File: topbin/coqobjinfo_bin.ml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+| Atom s -> Printf.fprintf chan "\"%s\"" (String.escaped s)
+| Node n -> Printf.fprintf chan "(%a)" print_list n
+
+and print_list chan = function
+| [] -> ()
+| [x] -> print chan x
+| [x; y] ->
+  Printf.fprintf chan "%a %a" print x print y
+| x :: rem ->
+  Printf.fprintf chan "%a %a" print x print_list rem
```
Comment (gares) [inline topbin/coqobjinfo_bin.ml:34]: maybe `(a b )` is also a valid sexp ;-)

File: topbin/coqobjinfo_bin.ml [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    let () = ObjFile.close_in ch in
+    raise e
+
+(* TODO: share me with Stateid *)
+type tasks = (int option, Obj.t) Stateid.request list * Obj.t
+
+let pr_request (req : (int option, Obj.t) Stateid.request) =
+  let open Stateid in
+  Sexp.Node [
+    Sexp.Atom req.kernel_name;
```
Comment (gares) [inline topbin/coqobjinfo_bin.ml:62]: @palmskog I think now the ball is yours, put here more prints for the `request` data type (the locs for example).
The `uuid : int option` is the place where the proof object is stored in `vio2vo`, I think you don't need it.

Comment (palmskog) [reply topbin/coqobjinfo_bin.ml:62]: @gares OK I will start to add prints, thanks. Indeed I'm not interested in `vio2vo` right now.

**Comment by silene** [comment]
As this is still a draft, I am postponing it to 8.15.

**Comment by gares** [comment]
Ok

**Pull request closed without merging**